sdk/identity/azure-identity/inc/azure/identity/workload_identity_credential.hpp

```diff
@@ -27,15 +27,34 @@ namespace Azure { namespace Identity {
    */
   struct WorkloadIdentityCredentialOptions final : public Core::Credentials::TokenCredentialOptions
   {
+    /**
+     * @brief The TenantID of the service principal. Defaults to the value of the environment
+     * variable AZURE_TENANT_ID.
+     */
+    std::string TenantId;
```

Comment:
How do you distinguish between the customer not providing a value (thus requesting AZURE_TENANT_ID) and the customer who wants to use an empty string for `TenantId`?

Or is `TenantId` required to have a value?

The same question applies to the other fields.

Reply:
As far as I can tell, empty strings for TenantId, ClientId, etc. are not considered valid, which is why we throw an exception in that case.

This is the same argument validation behavior in GoLang.